File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 10/08/2026, 02:37:27
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        int maxArea = 0;
        while(left<right){
            int left_height = height[left];
            int right_height = height[right];
            int min_height = min(left_height,right_height);
            maxArea = max(maxArea,min_height*(right-left));
            if (left_height < right_height)
                left++;
            else
                right--;
        }
        
        return maxArea;
    }
};
// https://youtu.be/ZHQg07n_tbg

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 10/08/2026, 02:37:21
class Solution(object):
    def maxArea(self, height):
        res = 0
        l, r = 0, len(height) - 1
        while l < r:
            area = (r - l) * min(height[l], height[r])
            res = max(res,area)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return res
        